consensus/crdt/consensus.go

```diff
@@ -428,7 +428,13 @@ func (css *Consensus) RmPeer(ctx context.Context, pid peer.ID) error {
 }
 
 // State returns the cluster shared state.
-func (css *Consensus) State(ctx context.Context) (state.ReadOnly, error) { return css.state, nil }
+func (css *Consensus) State(ctx context.Context) (state.ReadOnly, error) {
+	if css.state == nil {
+		return state.Empty(), nil
```

Comment:
I'd prefer if you set css.state to Empty in New during intialization. However this change seems to be completely unrelated to this PR?

Reply:
Also, I'm not even sure the WaitForSync in Join is necessary at this point. Maybe it can be removed in the stateless pintracker PR.

Reply:
Maybe an error should be returned. If someone calls State() once they get an empty state forever. The truth is the state is not setup so it should probably error.

Reply:
> Maybe an error should be returned. If someone calls State() once they get an empty state forever. The truth is the state is not setup so it should probably error.

Both consensus should behave similarly if we want to use them interchangeably. Currently, if state is not set, raft returns an empty state without error, which is why I did that same for crdt.


> However this change seems to be completely unrelated to this PR?

My intention was to fix a tests from panicking, so that build would pass. But this clearly opens more problems than fix. So yes, better to do it separately.

Reply:
> Both consensus should behave similarly if we want to use them interchangeably.

ok